### PR TITLE
fix: Branch name protection fixes

### PR DESCRIPTION
**Pull request by AlejandroFernandezLuces**
Fixes issues raised in https://github.com/ansys/pyansys-dev-guide/pull/528#issuecomment-2485834790

### CODE REVIEW

**Review by RobPasMue — APPROVED** [review]
Let's wait for @greschd

File: doc/source/_rulesets/branch_naming.json [hunk 1]
```diff
@@ -27,7 +27,7 @@
       "type": "branch_name_pattern",
       "parameters": {
         "operator": "regex",
-        "pattern": "^(feat|fix|chore|docs|style|refactor|test|testing|perf|ci|no-ci|build|dependabot|release|maint|junk|)\\/.*",
+        "pattern": "^(feat|fix|chore|docs|style|refactor|test|testing|perf|ci|no-ci|build|dependabot|release|maint|junk|chore|style|refactor|perf|ci|build)\\/.*",
```
Comment (RobPasMue) [inline doc/source/_rulesets/branch_naming.json:30]: I would rev
```suggestion
        "pattern": "^(feat|fix|chore|docs|style|refactor|test|testing|perf|ci|no-ci|build|dependabot|release|maint|junk|)\\/.*",
```
Already implemented them in my previous suggestion @AlejandroFernandezLuces - please verify next time

Comment (AlejandroFernandezLuces) [reply doc/source/_rulesets/branch_naming.json:30]: My bad... I added them to the contributing guide since we are enforcing them.

**Review by greschd — APPROVED** [review]
Thanks, I also like that the regex is now more understandable 🙂

**Pull request merged**